### PR TITLE
feat: add staff invitation saga for automatic provider account creation

### LINKED ISSUE #492: [TASK] Adding a team member automatically sends them a provider registration email
## Summary
When a business account adds a staff member via the Team section, the platform 
automatically sends an email to the provided email address inviting them to 
complete their registration. On clicking the link, they are registered as a 
Starter provider account, with any assigned programs already visible on 
their dashboard.

## User Story
As a business owner, I want adding a team member to automatically trigger 
an invitation so my staff can get set up on the platform without me having 
to walk them through a separate registration process.

## How It Works
1. Business account goes to Team > Add Staff Member
2. Enters staff member's name and email
3. Platform sends invitation email to that address
4. Staff member clicks link in email
5. Staff member completes their details (similar to standard registration)
6. Account is created as a Starter provider
7. Any programs already assigned to them by the business are 
   visible on their dashboard immediately

## Email Content
- "You've been invited to join [Business Name] on Klass Hero"
- Brief explanation of what Klass Hero is: ## Summary
When a business account adds a staff member via the Team section, the platform 
automatically sends an email to the provided email address inviting them to 
complete their registration. On clicking the link, they are registered as a 
Starter provider account, with any assigned programs already visible on 
their dashboard.

## Acceptance Criteria
- [ ] Adding a team member in Team section triggers invitation email automatically
- [ ] Email is sent to the email address provided by the business account
- [ ] Staff member completes registration via the link in the email
- [ ] Account is created as a Starter provider on completion
- [ ] Assigned programs are visible on staff member's dashboard immediately after registration

### PR DESCRIPTION
**Pull request by MaxPayne89**
Closes #492

## Summary

- Added a cross-context saga (Provider -> Accounts -> Provider) that automatically invites staff members to register when added with an email address. The saga uses integration events with Oban-backed durable delivery and compensating events for failure recovery.
- Implemented `StaffInvitation` LiveView at `/users/staff-invitation/:token` for token-based registration with pre-filled name/email, password validation, and mobile-responsive layout.
- Added `StaffDashboardLive` at `/staff/dashboard` with `:staff_provider` role routing and `Scope.resolve_roles` support for staff member role resolution.
- Extended provider dashboard team tab with invitation status badges, resend functionality with differentiated error messages, and emit failure compensation in `CreateStaffMember` and `ResendStaffInvitation` use cases.
- Hardened event payload handling: `normalize_keys/1` rescues unknown string keys, `StaffInvitationStatusHandler` uses `Map.fetch/2` instead of `Map.fetch!/2`, and all staff events are wired as critical handlers with idempotent status transitions.
- Added 80+ new tests covering domain model state machine, event handlers, use cases, saga integration (3 paths), LiveView flows, and compensation paths. Credo strict passes with 0 issues.

## Review Focus

- **Saga choreography and event flow** -- The saga spans two bounded contexts via 4 integration events (`staff_member_invited`, `staff_invitation_sent`, `staff_invitation_failed`, `staff_user_registered`). Review the event wiring in `config/config.exs:87-99` and handler implementations in `lib/klass_hero/accounts/adapters/driven/events/staff_invitation_handler.ex` and `lib/klass_hero/provider/adapters/driven/events/staff_invitation_status_handler.ex`.
- **Emit failure compensation** -- `CreateStaffMember` and `ResendStaffInvitation` persist to DB before emitting events. If emission fails, they compensate by transitioning the staff member to `:failed` status rather than leaving orphans in `:pending`. See `lib/klass_hero/provider/application/use_cases/staff_members/create_staff_member.ex:62-88` and `resend_staff_invitation.ex:31-57`.
- **DDD boundary enforcement** -- The `belongs_to :user` was replaced with `field :user_id, :binary_id` in `staff_member_schema.ex:33` to eliminate cross-context coupling. The `StaffInvitation` LiveView uses `Accounts.change_staff_registration/2` facade instead of importing `User` directly.
- **Token security** -- The `raw_token` flows through PubSub/Oban for the Accounts handler to construct the invitation URL. This is documented in `invitation_emitter.ex`. Tokens are SHA-256 hashed for storage, expire in 7 days, and are single-use (only valid when `invitation_status == "sent"`).
- **Scope extension** -- `Scope.resolve_roles/1` now conditionally queries for staff membership gated behind `if :staff_provider in (user.intended_roles || [])` to avoid a DB query for 99%+ of users. See `lib/klass_hero/accounts/scope.ex:57-60`.

## Test Plan

- [x] `mix precommit` passes (3,665 tests, 0 failures, compile --warnings-as-errors, format)
- [x] `mix credo --strict` reports 0 issues
- [x] Test-drive report: 20/20 checks pass (backend + UI + edge cases) -- `docs/reports/test-drive-2026-03-25.md`
- [x] Full saga verified end-to-end in dev: token generation -> invitation page -> registration -> event chain -> user_id linkage
- [x] Token reuse correctly shows "Invalid Invitation"
- [x] Mobile responsive layout verified at 375x667
- [ ] Verify invitation email content in `/dev/mailbox` after adding a staff member with email from the provider dashboard
- [ ] Verify the staff dashboard (`/staff/dashboard`) renders correctly when logged in as a staff user
- [ ] Verify German locale renders correctly on the staff invitation page

### CODE REVIEW

File: config/config.exs [hunk 1]
```diff
@@ -66,6 +66,10 @@ config :klass_hero, Oban,
 config :klass_hero, :accounts,
   for_storing_users: KlassHero.Accounts.Adapters.Driven.Persistence.Repositories.UserRepository
 
+# Base URL for constructing links in emails and event handlers
+# (avoids boundary violations from referencing KlassHeroWeb.Endpoint in domain code)
+config :klass_hero, :app_base_url, "http://localhost:4000"
```
[the remark this answers is not in the document]
Comment (MaxPayne89) [reply config/config.exs]: Addressed in adb07e25 — added `config :klass_hero, :app_base_url, "https://\#{host}"` in `runtime.exs` using the existing `PHX_HOST` env var. The `config.exs` default is kept for dev convenience (standard Phoenix practice).
